### PR TITLE
Use Hermes bytecode on Android

### LINKED ISSUE #1660: Use RN's Hermes JS engine for faster and leaner performance on Android
React Native's own JS engine, Hermes, is touting faster performance, smaller app footprint both for the JS bundle and the memory consumption.

It's be good if we switch to it, at least for Android. There are some instruction on [how to do it in the docs](https://facebook.github.io/react-native/docs/hermes/), and [this comment](https://github.com/facebook/hermes/issues/96#issuecomment-535393907) on a related ticket also seems useful with additional instructions.

### PR DESCRIPTION
**Pull request by hypest**
Fixes #1660

This PR is a continuation of https://github.com/wordpress-mobile/gutenberg-mobile/pull/2003, with a simpler approach and just using the bytecode version directly. If it proves more successful I will close the other PR.

Gutenberg PR: https://github.com/WordPress/gutenberg/pull/26732
WPAndroid PR: https://github.com/wordpress-mobile/WordPress-Android/pull/13297

I've left a comment with the (green!) [results of my Writing Flow manual tests here](https://github.com/WordPress/gutenberg/pull/26732#issuecomment-723043019).

#### Changes in this PR

* Use the bytecode version of the JS bundle for the apps and the tests

#### To test:
* No special testing other than all operations should look and feel faster.
* A perhaps easier to spot improvement is the editor startup time. Use the WPAndroid PR/build to start a new blog post and notice how much faster the editor comes online.

PR submission checklist:

- [x] I have considered adding unit tests where possible.
- [x] I have considered if this change warrants user-facing release notes [more info](https://github.com/wordpress-mobile/gutenberg-mobile/blob/develop/docs/Release-notes.md) and have added them to [RELEASE-NOTES.txt](https://github.com/wordpress-mobile/gutenberg-mobile/blob/develop/RELEASE-NOTES.txt) if necessary.

### CODE REVIEW

**Comment by mchowning** [comment]
Just noting that since these changes result in new `bundle/android/App.text.js` and `bundle/android/App.text.js.map` files, we should just make sure we generate and add those files to this PR before we merge it. I realize this comment is premature (it's only a draft PR after all), but I just wanted to note it so we didn't forget.

**Comment by hypest** [comment]
> Just noting that since these changes result in new bundle/android/App.text.js and bundle/android/App.text.js.map files, we should just make sure we generate and add those files to this PR before we merge it. I realize this comment is premature (it's only a draft PR after all), but I just wanted to note it so we didn't forget.

Not premature at all, thanks for the comment @mchowning! Not sure though, since we are not keeping the bundle files under source control anymore, what do you mean by generating and adding those files to this PR? For what is worth, I already added them to _.gitignore_, same as the non `...text...` ones.

**Comment by mchowning** [comment]
> since we are not keeping the bundle files under source control anymore, what do you mean by generating and adding those files to this PR? For what is worth, I already added them to .gitignore, same as the non ...text... ones.

I forgot that we're not committing the files anymore. Adding them to .gitignore is perfect. :+1:

**Comment by hypest** [comment]
I rebased this PR to remove the GB updates that are by [now merged](https://github.com/wordpress-mobile/gutenberg-mobile/pull/2784) into develop anyway and are not related to this PR.

File: package.json [hunk 1]
```diff
@@ -83,7 +85,9 @@
...
+    "test:e2e:bundle:android": "npm run test:e2e:bundle:android:text && npm run test:e2e:bundle:android:bytecode",
+    "test:e2e:bundle:android:text": "mkdir -p gutenberg/packages/react-native-editor/android/app/src/main/assets && npm run rn-bundle -- --reset-cache --platform android --dev false --minify false --entry-file index.js --bundle-output gutenberg/packages/react-native-editor/android/app/src/main/assets/index.android.text.bundle --assets-dest gutenberg/packages/react-native-editor/android/app/src/main/res",
+    "test:e2e:bundle:android:bytecode": "./gutenberg/node_modules/hermes-engine/`node -e \"const platform=require('os').platform();console.log(platform === 'darwin' ? 'osx-bin' : (platform === 'linux' ? 'linux64-bin' : (platform === 'win32' ? 'win64-bin' : 'unsupported-os')));\"`/hermes -emit-binary -O -out gutenberg/packages/react-native-editor/android/app/src/main/assets/index.android.bundle gutenberg/packages/react-native-editor/android/app/src/main/assets/index.android.text.bundle -output-source-map",
```
Comment (mchowning) [inline package.json:90]: Not an issue for this PR, but it feels like we should start extracting these out into some separate script files for the sake of readability. That might even open up some possibilities for reusing some of our npm script "logic" both here and in gutenberg/packages/react-native-editor.

**Review by mchowning — APPROVED** [review]
Tested these changes in the demo app and [WPAndroid](https://github.com/wordpress-mobile/WordPress-Android/pull/13297#pullrequestreview-525510919) and everything is working great.

File: package.json (hunk 1 above)
Comment (hypest) [reply package.json:90]: 💯
Non trivial command executions and pipelines, plus very long pathnames are impossible to read. I'd like to work on that, possibly even next week. Please, feel free to also share ideas on how to reuse logic 👍 !

**Pull request merged**